src/components/MainSidebar/ProgramDimensionsPanel/StageSelect.js

```diff
@@ -0,0 +1,39 @@
+import i18n from '@dhis2/d2-i18n'
+import { SingleSelect, SingleSelectOption } from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { acUpdateUiProgramStage } from '../../../actions/ui.js'
+import { sGetUiProgramStage } from '../../../reducers/ui.js'
+
+const STAGE_ALL = 'STAGE_ALL'
+
+const StageSelect = ({ optional, stages }) => {
+    const dispatch = useDispatch()
+    const selectedStageId = useSelector(sGetUiProgramStage)
+    const onChange = ({ selected }) =>
+        dispatch(acUpdateUiProgramStage(selected))
```

Comment:
This is a new pattern for us. We usually put the dispatch at the bottom in a `mapDispatchToProps`. Either way is fine I suppose, but its best if we keep a consistent code style. Is this the new preferred way? (so existing components should be refactored?

Reply:
same for `const selectedStageId = useSelector(sGetUiProgramStage)` of course